src/main/java/api/kirana/register/currencyexchange/service/CurrencyConversionServiceImpl.java

```diff
@@ -1,65 +0,0 @@
-package api.kirana.register.currencyexchange.service;
-
-import api.kirana.register.currencyexchange.models.ConversionResponse;
-import api.kirana.register.transactions.entity.Transactions;
-import api.kirana.register.transactions.repo.TransactionsDAO;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
-import org.springframework.web.client.RestTemplate;
-
-import java.util.Optional;
-@Service
-public class CurrencyConversionServiceImpl implements CurrencyConversionService{
-
-    @Autowired
-    private final TransactionsDAO transactionDAO;
-
-    @Autowired
-    public CurrencyConversionServiceImpl(TransactionsDAO transactionDAO) {
-        this.transactionDAO = transactionDAO;
-    }
-
-    private final RestTemplate restTemplate = new RestTemplate();
```

Comment:
We don't directly initialize resttemplate here, create a bean in appConfig class and autowire it here